### PR TITLE
Fixing Couchbase issues

### PR DESCRIPTION
**Pull request by shreekarSS**
Fixes https://github.com/red-hat-storage/ocs-ci/issues/2690

### CODE REVIEW

**Review by vasukulkarni — DISMISSED** [review]

**Review by vasukulkarni — DISMISSED** [review]

File: ocs_ci/ocs/pillowfight.py [hunk 1]
```diff
@@ -93,29 +94,31 @@ def run_pillowfights(self, replicas=1, num_items=None, num_threads=None):
...
-                            pf_completion_info = pf_status['terminated']['reason']
+                counter = 0
+                for pf_pod in pillowfight_pods:
+                    pod_info = self.up_check.exec_oc_cmd(
+                        f"get pods {pf_pod} -o json"
+                    )
+                    pf_status = pod_info['status']['containerStatuses'][0]['state']
+                    if 'terminated' in pf_status:
+                        pf_completion_info = pf_status['terminated']['reason']
+                        if pf_completion_info == 'Completed':
```
Comment (ebenahar) [inline ocs_ci/ocs/pillowfight.py:115]: use constants.STATUS_COMPLETED

File: tests/e2e/workloads/couchbase/test_couchbase_node_reboot.py [hunk 2]
```diff
@@ -49,34 +53,56 @@ def cb_setup(self, couchbase_factory_fixture):
...
-            node_list = get_master_nodes()
+            master_node = get_typed_nodes(pod_name_of_node, num_of_nodes=1)
 
-        node_1 = get_node_objs(
-            node_list[random.randint(0, len(node_list) - 1)])
+        # Restart relevant node
+        if pod_name_of_node == 'master':
+            nodes.restart_nodes(master_node, wait=False)
+            waiting_time = 40
+            log.info(f"Waiting {waiting_time} seconds...")
```
Comment (ebenahar) [inline tests/e2e/workloads/couchbase/test_couchbase_node_reboot.py:80]: why are we waiting here? if it's for the nodes to come up to Ready state, we have a function for that in node.py - wait_for_nodes_status

File: tests/e2e/workloads/couchbase/test_couchbase_node_reboot.py [hunk 3]
```diff
@@ -49,34 +53,56 @@ def cb_setup(self, couchbase_factory_fixture):
...
+            time.sleep(waiting_time)
+        else:
+            node_1 = get_node_objs(node_list[random.randint(0, len(node_list) - 1)])
```
Comment (ebenahar) [inline tests/e2e/workloads/couchbase/test_couchbase_node_reboot.py:83]: is there any other var like `node_2`? if not, shouldn't we call the var `node`?

File: ocs_ci/ocs/pillowfight.py (hunk 1 above)
Comment (shreekarSS) [reply ocs_ci/ocs/pillowfight.py:115]: ack

File: tests/e2e/workloads/couchbase/test_couchbase_node_reboot.py (hunk 2 above)
Comment (shreekarSS) [reply tests/e2e/workloads/couchbase/test_couchbase_node_reboot.py:80]: I referred this https://github.com/red-hat-storage/ocs-ci/blob/master/tests/e2e/workloads/amq/test_amq_node_reboot_and_shutdown.py#L104
in master node reboot, we don't have checks for checking cluster connectivity in restart_nodes()
So I skipped wait in restart_nodes(), and kept wait_for_nodes_status() and wait_for_cluster_connectivity() after restart.

File: tests/e2e/workloads/couchbase/test_couchbase_node_reboot.py (hunk 3 above)
Comment (shreekarSS) [reply tests/e2e/workloads/couchbase/test_couchbase_node_reboot.py:83]: No node_2,  will update

File: tests/e2e/workloads/couchbase/test_couchbase_node_reboot.py (hunk 2 above)
Comment (ebenahar) [reply tests/e2e/workloads/couchbase/test_couchbase_node_reboot.py:80]: you can use ocs_ci.ocs.ocp.wait_for_cluster_connectivity

Comment (shreekarSS) [reply tests/e2e/workloads/couchbase/test_couchbase_node_reboot.py:80]: Yes , I'm using it

**Review by vasukulkarni — APPROVED** [review]

File: tests/e2e/workloads/couchbase/test_couchbase_node_reboot.py (hunk 2 above)
Comment (ebenahar) [reply tests/e2e/workloads/couchbase/test_couchbase_node_reboot.py:80]: but nodes.restart_nodes() has a wait functionality - ocs_ci.ocs.node.wait_for_nodes_status

What am I missing?

Comment (shreekarSS) [reply tests/e2e/workloads/couchbase/test_couchbase_node_reboot.py:80]: Rebooting master sometimes it fails with an unauthorized error. So not waiting there and in below lines trying to catch ResourceWrongStatusException errors and untill node becomes ready state. So using  wait_for_nodes_status() after some sleep.

**Review by ebenahar — APPROVED** [review]

**Pull request merged**